### PR TITLE
feat(ops): get_wal_archive_status — WAL archive health (roadmap 5.2)

### PR DESCRIPTION
**Pull request by devopam**
## Summary

Realises roadmap **5.2**. A new read-only tool `get_wal_archive_status` that surfaces WAL-archiving health from `pg_stat_archiver` + the archive-mode GUCs. Companion to `read_pg_wal_records` (2.3, which inspects WAL *records*); this covers the WAL *archive*.

## Why

When `archive_command` / `archive_library` starts failing (full archive volume, bad object-store credentials, network partition to the object store), Postgres retries the **same** segment forever and WAL accumulates in `pg_wal/` until the volume fills — a silent, slow-motion outage. The early signal lives in `pg_stat_archiver`: a rising `failed_count` and a `last_failed_time` more recent than `last_archived_time`. This tool makes that a one-call check with a computed `healthy` verdict + human-readable `detail`.

## Behaviour

- `healthy = false` when archiving is enabled AND the latest attempt failed (`last_failed_time` newer than `last_archived_time`, or failures with nothing ever archived).
- Disabled archiving (`archive_mode = off`) → `healthy = true` (nothing to monitor); `always` / `on` both count as enabled.
- Read-only; never raises — driver failure / empty result → `available = false` with the error in `detail`.

## Security

The `archive_command` string is **not** echoed (it can embed object-store credentials) — only a boolean `archive_command_set`.

## Test plan

- [x] `uv run pytest tests/unit/test_wal_archive.py` — 9/9 pass (healthy / failing / recovered / disabled / always-mode / no-archives-ever / driver-failure / empty / command-redaction)
- [x] Typed dataclass return → emits an `outputSchema`; added to the manifest (floor 192 → 193)
- [x] Surface + return-shape snapshots regenerated
- [x] Full suite: 2514 passed, 110 skipped
- [x] `ruff` + `mypy` clean

## Roadmap linkage

Advances roadmap row: 5.2

---
_Generated by [Claude Code](https://claude.ai/code/session_0122yLZLJ8t4W43sdN6BmTZc)_

## Summary by Sourcery

Add a new read-only WAL archiving health probe and wire it into the tools surface, capability routing, and structured-output manifest.

New Features:
- Introduce the get_wal_archive_status tool to report WAL archiving health from pg_stat_archiver and related configuration, returning a structured WalArchiveStatus payload.

Enhancements:
- Classify get_wal_archive_status in the operations_and_health capability bucket and expose it through the server tool registry.

Documentation:
- Document the new get_wal_archive_status WAL archiving health probe in the changelog and mark roadmap item 5.2 as shipped in the feature shortlist.

Tests:
- Add unit tests and contract snapshots to verify get_wal_archive_status behavior, output schema, and handling of error and edge conditions.

### CODE REVIEW

File: src/mcpg/wal_archive.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    if latest_failed_more_recent:
+        return False, (
+            f"WAL archiving is FAILING — {failed_count} failure(s); the most recent attempt failed. "
+            "Check the archive_command / archive_library target (full volume, credentials, "
+            "or network to the object store). WAL will accumulate in pg_wal/ until resolved."
+        )
+    return True, (
+        f"WAL archiving recovered — {failed_count} historical failure(s) but the latest "
+        f"attempt succeeded ({archived_count} segment(s) archived total)."
+    )
```
[the remark this answers is not in the document]
Comment (devopam) [reply src/mcpg/wal_archive.py:104]: Fixed in 279a636. `_assess` now takes a `latest_failed_more_recent: bool` and no longer compares timestamp text — the chronological decision is made in SQL on the native `timestamptz` columns (see the other two threads).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/mcpg/wal_archive.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+            "  s.archived_count, "
+            "  s.last_archived_wal, "
+            "  s.last_archived_time::text AS last_archived_time, "
+            "  s.failed_count, "
+            "  s.last_failed_wal, "
+            "  s.last_failed_time::text AS last_failed_time, "
+            "  s.stats_reset::text AS stats_reset "
+            "FROM pg_stat_archiver s",
+            force_readonly=True,
+        )
```
[the remark this answers is not in the document]
Comment (devopam) [reply src/mcpg/wal_archive.py:129]: Done in 279a636 — added `COALESCE(s.last_failed_time > s.last_archived_time, s.last_failed_time IS NOT NULL) AS latest_failed_more_recent` to the SELECT.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/mcpg/wal_archive.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+    last_archived_time = c.get("last_archived_time")
+    last_failed_time = c.get("last_failed_time")
+
+    healthy, detail = _assess(
+        archiving_enabled=archiving_enabled,
+        archived_count=archived_count,
+        failed_count=failed_count,
+        last_archived_time=last_archived_time,
+        last_failed_time=last_failed_time,
+    )
```
[the remark this answers is not in the document]
Comment (devopam) [reply src/mcpg/wal_archive.py:189]: Done in 279a636 — the handler reads `c.get("latest_failed_more_recent")` and keeps the text-comparison fallback for callers/test fakes that don't project the column.

---
_Generated by [Claude Code](https://claude.ai/code)_
